### DISCUSSION TITLE
Getting error while  adding org roles and space roles to users

### DISCUSSION THREAD

**Discussion by ashjadhav1** [SAP/terraform-provider-cloudfoundry#200] [Q&A]
Hi we have tried to add org roles and space roles to users using cloudfoundry_org_role and cloudfoundry_space_role.
but it is giving below error -

Error: API Error Registering Role
│ 
│   with cloudfoundry_space_role.sp-test-usr,
│   on testusr.tf line 131, in resource "cloudfoundry_space_role" "sp-test-usr":
│  131:   resource "cloudfoundry_space_role" "sp-test-usr" {
│ 
│ Could not register Role with user ID  : executing POST request for
│ /v3/roles failed: cfclient error (CF-UnprocessableEntity|10008): Users
│ cannot be assigned roles in a space if they do not have a role in that
│ space's organization.

We have tried few scenarios for the same.
1. Assigning role to user in org and space with default origin(idp)
    In this scenario user assigned role with org_user and not able to assign role space.
2. Assigning role to users in org and space with custom and default origin(idp) while creating new subaccount.
     In this scenario we are able to assign roles to users in both org and space. But when we tried same thing with other user in same 
     subaccount it is giving error.

Please assist on this as we see its working but not fully.

**Comment by Dray56** [comment]
Hi @ashjadhav1,

Kindly refer https://github.com/SAP/terraform-provider-cloudfoundry/issues/156 . Hope it fixes the issue. 

For further assistance please raise future issues [here](https://github.com/cloudfoundry/terraform-provider-cloudfoundry) since this provider is now deprecated and no longer maintained.
